substrate/frame/support/src/lib.rs

```diff
@@ -781,6 +783,34 @@ macro_rules! assert_error_encoded_size {
 	} => {};
 }
 
+/// Do something hypothetically by rolling back any changes afterwards.
+///
+/// Returns the original result of the closure.
+#[macro_export]
+#[cfg(feature = "experimental")]
+macro_rules! hypothetically {
+	( $e:expr ) => {
+		$crate::storage::transactional::with_transaction(
+							|| -> $crate::__private::TransactionOutcome<Result<_, $crate::__private::DispatchError>> {
```

Comment:
weirdly formatted, love the idea though.

Reply:
I changed it back but the formatter seems to modify it 🤔 Its less bad than before at least.